api/v1/reviewers/recommendations/ReviewerRecommendationController.php

```diff
@@ -0,0 +1,219 @@
+    {
+        $illuminateRequest = $args[0]; /** @var \Illuminate\Http\Request $illuminateRequest */
+        $actionName = static::getRouteActionName($illuminateRequest);
+
+        $this->addPolicy(new UserRolesRequiredPolicy($request), true);
+        $this->addPolicy(new ContextAccessPolicy($request, $roleAssignments));
+
+        if (in_array($actionName, ['edit', 'updateStatus', 'delete'])) {
+            $this->addPolicy(
+                new RecommendationAccessPolicy(
+                    $request,
+                    static::getRequestedRoute($illuminateRequest)->parameter('reviewerRecommendationId')
+                )
+            );
+        }
+
+        return parent::authorize($request, $args, $roleAssignments);
+    }
+
+    /**
+     * @copydoc \PKP\core\PKPBaseController::getGroupRoutes()
+     */
+    public function getGroupRoutes(): void
+    {
+        Route::get('{reviewerRecommendationId}', $this->get(...))
+            ->name('reviewer.recommendations.get')
+            ->whereNumber(['reviewerRecommendationId']);
+
+        Route::get('', $this->getMany(...))
+            ->name('reviewer.recommendations.getMany');
+
+        Route::post('', $this->add(...))
+            ->name('reviewer.recommendations.add');
+
+        Route::put('{reviewerRecommendationId}', $this->edit(...))
+            ->name('reviewer.recommendations.edit')
+            ->whereNumber(['reviewerRecommendationId']);
+
+        Route::put('{reviewerRecommendationId}/status', $this->updateStatus(...))
+            ->name('reviewer.recommendations.edit.status')
+            ->whereNumber(['reviewerRecommendationId']);
+
+        Route::delete('{reviewerRecommendationId}', $this->delete(...))
+            ->name('reviewer.recommendations.delete')
+            ->whereNumber(['reviewerRecommendationId']);
+    }
+
+    /**
+     * Get specific recommendation response
+     */
+    public function get(Request $illuminateRequest): JsonResponse
+    {
+        $recommendation = ReviewerRecommendation::find($illuminateRequest->route('reviewerRecommendationId'));
+
+        return response()->json(
+            (new ReviewerRecommendationResource($recommendation))->toArray($illuminateRequest),
+            Response::HTTP_OK
+        );
+    }
+
+    /**
+     * Get all recommendations response
+     */
+    public function getMany(Request $illuminateRequest): JsonResponse
+    {
+        $recommendations = ReviewerRecommendation::query()
+            ->withContextId(Application::get()->getRequest()->getContext()->getId())
+            ->get();
+
+        return response()->json([
+            'items' => ReviewerRecommendationResource::collection($recommendations),
+            'itemMax' => $recommendations->count(),
+        ], Response::HTTP_OK);
+    }
+
+    /**
+     * Add new recommendation
+     */
+    public function add(AddReviewerRecommendation $illuminateRequest): JsonResponse
+    {
+        $validateds = $illuminateRequest->validated();
```

Comment:
Typo in variable name. Also, if the data is not validated, do we manually need to return a different response here?

Reply:
> Typo in variable name

Is this for `$validateds` ? this is intentionally done as plural from of `validated` so identify as all validated data .

> if the data is not validated, do we manually need to return a different response here?

As we are using `FromRequest` as `AddReviewerRecommendation`, so any failed validation will be handled there and return back from there without reaching to controller code . Here we will get validated data after successful validation, so no need to return different response as will never reach here without completing the validation successfully .

Reply:
Gotcha, I thought that was probably the case but I didn't know if how we were using it would handle all that automatically.